src/track/beats.cpp

```diff
@@ -21,10 +21,48 @@ struct BeatGridV1Data {
 
 constexpr double kEpsilon = 0.01;
 
+// The amount that `Beats::tryAdjustTempo()` changes the last marker's BPM by.
+constexpr double kBpmAdjustStep = 0.01;
+
+int roundBeatCountToFullBar(int numBeats, int beatsPerBar) {
+    const int numExcessBeats = (numBeats % beatsPerBar);
+    if (numExcessBeats < (beatsPerBar / 2)) {
+        return numBeats - numExcessBeats;
+    }
+
+    return numBeats + beatsPerBar - numExcessBeats;
+}
+
+int calculateBeatsTillNextMarker(const mixxx::Beats::ConstIterator& it,
+        mixxx::audio::FramePos nextMarkerPosition) {
+    const double numBeatsDouble = std::fabs(nextMarkerPosition - *it) / it.beatLengthFrames();
+    const int numBeatsInt = static_cast<int>(std::round(numBeatsDouble));
```

Comment:
We round here twice. That may lead to a wrong overall rounding

Reply:
Assuming every `int` can safely be represented as `double`, which is the case on the overwhelming majority of desktop platforms where `int` is 32-bit and `double` is an 64-bit IEEE-754 floating point and given that `std::round` is guaranteed to round to the _nearest_ integer, this is usually safe. See https://stackoverflow.com/a/47153373/19890279.

I agree, however, that it might be more elegant to go with `std::lround`. We would still have to cast the `long` to an `int` though.